crypto/x509/utils/src/lib.rs

```diff
@@ -0,0 +1,292 @@
+// Copyright (c) 2018-2020 MobileCoin Inc.
+
+//!
+
+use mc_crypto_keys::Ed25519Public;
+use std::{cmp, convert::TryFrom, io::Cursor, str::FromStr};
+use x509_parser::{
+    certificate::X509Certificate, der_parser::oid::Oid, pem::Pem, x509::AlgorithmIdentifier,
+};
+
+/// An iterator of [`Pem`] structures created over a string slice.
+pub struct PemStringIter<T> {
+    string: Cursor<T>,
+    offset: usize,
+}
+
+impl<T: AsRef<[u8]>> PemStringIter<T> {
+    /// Create a new iterator based on the given string.
+    fn new(inner: T) -> Self {
+        Self {
+            string: Cursor::new(inner),
+            offset: 0,
+        }
+    }
+}
+
+impl<T: AsRef<[u8]>> Iterator for PemStringIter<T> {
+    type Item = Pem;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        Pem::read(&mut self.string)
+            .map(|(pem, new_offset)| {
+                self.offset = new_offset;
+                pem
+            })
+            .ok()
+    }
+}
+
+/// A trait used to monkey-patch a pem parsing iterator over string slices
+pub trait PemStringIterable: AsRef<[u8]> + Sized {
+    /// Create an iterator over a string which contains Pem objects
+    fn into_pem_iter(self) -> PemStringIter<Self>;
+}
+
+/// Anything which can be referenced as a byte slice gets into_iter_pem()
+impl<T: AsRef<[u8]>> PemStringIterable for T {
+    fn into_pem_iter(self) -> PemStringIter<T> {
+        PemStringIter::new(self)
```

Comment:
nit: this could be the default implementation of the function, in the trait definition